### PR TITLE
Arnej/global filter after fetch postings

### PR DESCRIPTION
**Pull request by arnej27959**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

@geirst @toregge @havardpe please review

### CODE REVIEW

**Review by havardpe — COMMENTED** [review]
Not sure what happens when the strictness of fetch postings and create iterator do not match...

File: searchlib/src/vespa/searchlib/queryeval/nearest_neighbor_blueprint.cpp [hunk 1]
```diff
@@ -84,10 +87,15 @@ NearestNeighborBlueprint::set_global_filter(const GlobalFilter &global_filter)
...
+    LOG(debug, "set_global_filter with: %s / %s / %s",
+        (_approximate ? "approximate" : "exact"),
+        (nns_index ? "nns_index" : "no_index"),
+        (_global_filter->has_filter() ? "has_filter" : "no_filter"));
     if (_approximate && nns_index) {
         uint32_t est_hits = _attr_tensor.getNumDocs();
         if (_global_filter->has_filter()) {
             uint32_t max_hits = _global_filter->filter()->countTrueBits();
+            LOG(debug, "set_global_filter getNumDocs: %u / max_hits %u", est_hits, max_hits);
             if (max_hits * 10 < est_hits) {
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/queryeval/nearest_neighbor_blueprint.cpp:99]: Until we have data on performance tests, perhaps we should _not_ do fallback to brute force. That will make it harder to interpret the results.

**Review by geirst — APPROVED** [review]
👍

**Comment by hmusum** [comment]
Seems like system tests tests/search/explain/explain.rb started failing after this PR, you might want to have a look

**Pull request merged**